### PR TITLE
Empty response

### PR DESCRIPTION
**Pull request by flippinjoe21**
This PR adds the ability to specify a route that returns nil as success

### CODE REVIEW

File: Pod/Classes/RouteProcessor.swift [hunk 1]
```diff
@@ -4,6 +4,7 @@ import Foundation
 public enum RouteResult {
     case Array([[String: AnyObject]])
     case Dict([String: AnyObject])
+    case Empty
     case Unknown
     
     public var data: AnyObject? {
```
Comment (ndis1) [inline Pod/Classes/RouteProcessor.swift]: why is this `AnyObject` not `Any` ?

File: Pod/Classes/APIRouter.swift [hunk 2]
```diff
@@ -112,6 +112,37 @@ public extension APIRouter {
             })
         }
         
+        runRouteWithBeforeHandlers(route, query: query, body: body, responder: responder)
+    }
+    
+    public func call<T: APIRouteable where T.ArgumentType == AnyObject?, T.SerializedType == AnyObject?>(route: T, query: [String: AnyObject]? = nil, body: AnyObject? = nil, handler: Result<T.SerializedType> -> Void) {
```
Comment (ndis1) [inline Pod/Classes/APIRouter.swift]: this pattern could get matched by someone who wanted an argument and serialized type `AnyObject` - how about a serialized type `EmptyResponse` ?

**Comment by ndis1** [comment]
I'm not thrilled about having a serializer - even an empty one - when we are not doing any serializing. Have you explored having a different implementer of APIRespondable that doesn't have a serializer?

**Comment by flippinjoe21** [comment]
Yeah.    I wasn't sure if we wanted to use the serializer or  not.   I mainly just left it in place because that's how everything works now.   I didn't know if there was any case where we wanted to potentially turn the nil response into an empty object or something

**Comment by jgrandelli** [comment]
The only thing we will want to make sure we have on the empty response is the ability to check for success. Which we could probably do by with just the result type where value is nil and error is either nil or not

**Comment by jgrandelli** [comment]
I do agree though that having different responsible just for the empty case is kind bunk

File: Example/Tests/TestHelpers.swift [hunk 3]
```diff
@@ -9,11 +9,16 @@
...
+    case FileNotFound
+}
+
+func dataFromFile(path: String, ext: String = "json") throws -> AnyObject?  {
     if let path = NSBundle.mainBundle().pathForResource(path, ofType: ext), d = NSData(contentsOfFile: path) {
         return try? NSJSONSerialization.JSONObjectWithData(d, options: [])
     }
-    return nil
+    
+    throw DataError.FileNotFound
```
Comment (ndis1) [inline Example/Tests/TestHelpers.swift]: nice

**Comment by ndis1** [comment]
+1 big improvement over what we have now

short term - if we end up writing a ton of these we'll probably end up writing some convenience to avoid specifying the `empty` serializer for each route. Might also wrap the `APIRoute<EmptyResponse, NoResponseType>` in a struct so we don't have to write out that generic all over the place

long term - would be nice to have a router (base protocol?) that didn't care about serialization at all

**Pull request merged**